code-editor/scripts/on-start.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -ex
+ASI_VERSION=0.3.0
+
+# PARAMETERS 
+IDLE_TIME=3600  # in seconds, change this to desired idleness time before app shuts down
+
+AUTOSTOP_FILE_URL="https://github.com/aws-samples/sagemaker-studio-apps-lifecycle-config-examples/releases/download/sagemaker_code_editor_auto_shut_down-$ASI_VERSION.tar.gz"
+
+echo "Fetching the autostop script"
+echo "Downloading the autostop script package"
+
+# Download asset from corresponding repo release
+curl -L -o sagemaker_code_editor_auto_shut_down-$ASI_VERSION.tar.gz $AUTOSTOP_FILE_URL
+
+echo "Extracting the package"
+tar -xvzf "sagemaker_code_editor_auto_shut_down-$ASI_VERSION.tar.gz"
+
+# Moving the autostop.py to the current directory
+# Adjust the path according to the structure inside the tar.gz file
+echo "Moving the autostop.py to the current directory"
+mv sagemaker_code_editor_auto_shut_down/python-package/src/sagemaker_code_editor_auto_shut_down/autostop.py ./
+
+echo "Detecting Python install with boto3 install"
+sudo apt-get update -y
+sudo apt-get install -y vim
+sudo sh -c 'printf "#!/bin/sh\nexit 0" > /usr/sbin/policy-rc.d'
+sudo apt-get install -y cron
+sudo service cron start
```

Comment:
This should not be required given the patch on line 27.

Reply:
That's a good point - the line `sudo service cron start` is usually needed for Linux environments to ensure the `cron` service is running for scheduled tasks

But in containerized environments or where a workaround like `sudo sh -c 'printf "#!/bin/sh\nexit 0" > /usr/sbin/policy-rc.d'` is used to prevent automatic service management, starting the cron service explicitly may not be required and can be omitted - will try testing it before committing 

(link: https://manpages.debian.org/unstable/init-system-helpers/invoke-rc.d.8.en.html)

Reply:
Update: verified via testing, not needed and removed